docker/datahub-ingestion/Dockerfile-slim-only

```diff
@@ -0,0 +1,25 @@
+# Defining environment
+ARG BASE_IMAGE=acryldata/datahub-ingestion-base
+ARG DOCKER_VERSION=latest
+
+FROM $BASE_IMAGE:$DOCKER_VERSION as base
+USER 0
+
+COPY ./metadata-ingestion /datahub-ingestion
+
+ARG RELEASE_VERSION
+WORKDIR /datahub-ingestion
+RUN sed -i.bak "s/__version__ = \"0.0.0.dev0\"/__version__ = \"$RELEASE_VERSION\"/" src/datahub/__init__.py && \
+    cat src/datahub/__init__.py && \
+    chown -R datahub /datahub-ingestion
```

Comment:
I was really hoping to avoid having lots of dockerfiles - is it possible to do this stuff using build args?

Reply:
lack of support for target in the gradle docker plugin, if i merge then it'll build more then it needs and takes longer for the spark smoke test

Reply:
This is for a specific instance, I don't think it's indicative of a proliferation of Dockerfiles in general, so I think it's probably okay. If we start needing slim images of others as well we can think about getting tricky with build args, profiles, etc. is my two cents 😄